File: sources/net.sf.j2s.core/test/dev/j2s/java/text/MessageFormat.js
Clazz.declarePackage ("java.text");

(function(){var C$ = Clazz.decorateAsClass (function () {
Clazz.newInstance$ (this, arguments);
}, java.text, "MessageFormat");

Clazz.newMethod$(C$, '$init$', function () {
this.pattern = null;
}, 1);

Clazz.newMethod$ (C$, 'construct$S', function (pattern) {
C$.$init$.apply(this);
this.pattern = pattern;
}, 1);

Clazz.newMethod$ (C$, 'construct$S$java_util_Locale', function (pattern, locale) {
C$.$init$.apply(this);
this.pattern = pattern;
}, 1);

Clazz.newMethod$ (C$, 'format$S$OA', function (pattern, args) {
return pattern.replace (/\{(\d+)\}/g, function ($0, $1) {
var i = parseInt ($1);
if (args == null) return null;
return args[i];
});
}, 1);

Clazz.newMethod$ (C$, 'format$O', function (obj) {
return java.text.MessageFormat.format$S$OA (this.pattern,  Clazz.newArray (-1, [obj]));
});

Clazz.newMethod$(C$, 'construct', function () {Clazz.super$(C$, this);
C$.$init$.apply(this);
},true);
})()

//Created 2017-08-08 06:13:45

File: sources/net.sf.j2s.java.core/src_4.2/java/text/MessageFormat.java
/*******************************************************************************
 * Copyright (c) 2007 java2script.org and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     Zhou Renjian - initial API and implementation
 *******************************************************************************/

package java.text;

import java.util.Locale;


/**
 * @author zhou renjian
 *
 * 2006-10-10
 */
public class MessageFormat {
	private String pattern;
	
    public MessageFormat(String pattern) {
    	this.pattern = pattern;
    }
    
    public MessageFormat(String pattern, Locale locale) {
    	this.pattern = pattern;
    }

    /**
     * @j2sNativeSrc
     * return pattern.replace (/\{(\d+)\}/g, function ($0, $1) {
     * 	var i = parseInt ($1);
     * 	if (args == null) return null;
     *  return args[i];
     * });
     * @j2sNative
     * return a.replace (/\{(\d+)\}/g, function ($0, $1) {
     * 	var i = parseInt ($1);
     * 	if (b == null) return null;
     *  return b[i];
     * });
     */ 
    public static String format(String pattern, Object[] args) {
    	return pattern;
    }
    
    public final String format (Object obj) {
    	return format(pattern, new Object[] { obj });
    }
}
